### ISSUE TITLE
2022 2학기 빅데이터 과제 이미지

### ISSUE THREAD

**Issue by physics08** [physics08/physics08#4]
![image](https://user-images.githubusercontent.com/105033906/194721628-709ef5f5-fa25-47bc-90ec-6bf9d68845bf.png)
from sklearn.datasets import make_blobs
X,y =make_blobs(n_samples=210, n_features=2, centers=3, cluster_std=0.5, shuffle=True, random_state=0)

import matplotlib.pyplot as plt
plt. scatter(X[:,0], X[:,1], c='white', marker='o', edgecolor='black', s=50)

plt.grid()
plt.tight_layout()
plt.show()

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/194721652-aed4e7ed-c9e1-4afe-bfd1-21ab83ab52b1.png)
plt.scatter(X[y_km==0,0], X[y_km==0,1], s=50, c='lightgreen', marker='s', edgecolor='black', label='cluster1')
plt.scatter(X[y_km==1,0], X[y_km==1,1], s=50, c='orange', marker='o', edgecolor='black', label='cluster2')
plt.scatter(X[y_km==2,0], X[y_km==2,1], s=50, c='lightblue', marker='v', edgecolor='black', label='cluster3')
plt.scatter(km.cluster_centers_[:,0], km.cluster_centers_[:,1], s=250, marker='*', c='red', edgecolor='black', label='Centroids')



plt.legend(scatterpoints=1)
plt.grid()
plt.tight_layout()
plt.show()

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/194721676-3702a135-42e1-4c03-bb4e-363ea4eefb4d.png)


distortions=[]
for i in range(1,11):
  km=KMeans(n_clusters=i, init='random', n_init=10, max_iter=300, random_state=0)
  km.fit(X)
  distortions.append(km.inertia_)

plt.plot(range(1,11), distortions, marker='o')
plt.xlabel('Number of clusters')
plt.ylabel('Distortion')
plt.tight_layout()
plt.show()

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/194721694-83b4e133-d5a7-4193-87d7-641167c79f1b.png)


km=KMeans(n_clusters=3, init='random', n_init=10, max_iter=300, tol=1e-04, random_state=0)
y_km=km.fit_predict(X)

import numpy as np
from matplotlib import cm
from sklearn.metrics import silhouette_samples

cluster_labels=np.unique(y_km)
n_clusters=cluster_labels.shape[0]
silhouette_vals=silhouette_samples(X, y_km, metric='euclidean')

y_ax_lower, y_ax_upper=0,0
yticks=[]
for i, c in enumerate(cluster_labels):
  c_silhouette_vals=silhouette_vals[y_km==c]
  c_silhouette_vals.sort()
  y_ax_upper+=len(c_silhouette_vals)
  color=cm.jet(float(i)/n_clusters)
  plt.barh(range(y_ax_lower, y_ax_upper), c_silhouette_vals, height=1.0, edgecolor='none', color=color)
  yticks.append((y_ax_lower+y_ax_upper)/2.0)
  y_ax_lower+=len(c_silhouette_vals)
silhouette_avg=np.mean(silhouette_vals)
plt.axvline(silhouette_avg, color='red', linestyle='--')
plt.yticks(yticks, cluster_labels+1)
plt.ylabel('Cluster')
plt.xlabel('silhouette coefficient')
plt.tight_layout()
plt.show()

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/194721701-1d22518d-fa42-4335-8ae7-13eedeecce93.png)

from sklearn.cluster import KMeans
km=KMeans(n_clusters=2, init='random', n_init=10, max_iter=300, tol=1e-04, random_state=3)
y_km=km.fit_predict(X)



plt.scatter(X[y_km==0,0], X[y_km==0,1], s=50, c='lightgreen', marker='s', edgecolor='black', label='cluster1')
plt.scatter(X[y_km==1,0], X[y_km==1,1], s=50, c='orange', marker='o', edgecolor='black', label='cluster2')
plt.scatter(km.cluster_centers_[:,0], km.cluster_centers_[:,1], s=250, marker='*', c='red', edgecolor='black', label='Centroids')



plt.legend(scatterpoints=1)
plt.grid()
plt.tight_layout()
plt.show()

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/194721712-a55b471e-5b96-4bfa-b1b1-bbfaa3bcc96e.png)


cluster_labels=np.unique(y_km)
n_clusters=cluster_labels.shape[0]
silhouette_vals=silhouette_samples(X, y_km, metric='euclidean')

y_ax_lower, y_ax_upper=0,0
yticks=[]
for i, c in enumerate(cluster_labels):
  c_silhouette_vals=silhouette_vals[y_km==c]
  c_silhouette_vals.sort()
  y_ax_upper+=len(c_silhouette_vals)
  color=cm.jet(float(i)/n_clusters)
  plt.barh(range(y_ax_lower, y_ax_upper), c_silhouette_vals, height=1.0, edgecolor='none', color=color)
  yticks.append((y_ax_lower+y_ax_upper)/2.0)
  y_ax_lower+=len(c_silhouette_vals)
silhouette_avg=np.mean(silhouette_vals)
plt.axvline(silhouette_avg, color='red', linestyle='--')
plt.yticks(yticks, cluster_labels+1)
plt.ylabel('Cluster')
plt.xlabel('silhouette coefficient')
plt.tight_layout()
plt.show()

**Comment by physics08** [comment]
![6](https://user-images.githubusercontent.com/105033906/194722516-5b1c5917-2019-4cc9-af8f-f68b8b116f46.png)
![1](https://user-images.githubusercontent.com/105033906/194722519-601525a9-5137-4def-94cf-30a0f8f89b1d.png)
![2](https://user-images.githubusercontent.com/105033906/194722520-4bb532b7-1bdf-442b-91d0-c7ddcdb03a05.png)
![3](https://user-images.githubusercontent.com/105033906/194722521-ce6eb189-b6dd-49eb-a042-e48cd8010bda.png)
![4](https://user-images.githubusercontent.com/105033906/194722524-da60c0fe-5f80-4e4f-9bc3-8d6da864a149.png)
![5](https://user-images.githubusercontent.com/105033906/194722525-d3076625-8fc8-4687-b29a-4b6778de9fcf.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/194722958-304a1a0c-3506-43c6-8daf-b527a59fdb0e.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/194736194-0365bce6-5a1a-4447-b900-8395c0f6cdcf.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/194736357-fce47de1-e275-4580-b895-4611459733de.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/194736583-230f2d9a-160e-40f3-8825-7d6887886416.png)

**Comment by physics08** [comment]
![arrow](https://user-images.githubusercontent.com/105033906/194739160-32c734a0-fa27-4aa0-8f79-963c2c8bffe2.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/194971921-9c03ed30-3526-4ca9-8960-bb86e9239d81.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195310415-ff450e8e-63b6-4fd1-912b-2b223cb52542.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195310486-6055be26-f221-4626-ac03-c114515a7d8e.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195310510-11930eb9-5b9f-4769-903a-cf6ff66988f4.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195310556-102130cc-9aa2-4b8e-8f12-e91281c9da23.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195310596-0fed0fe1-475d-45b7-a134-4cd21f40c2a9.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195310638-2360e105-82fe-4850-a5d3-6565795dccef.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195310667-fe720b95-1db2-4147-9e68-430ca0bab66c.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195312243-2a51be5c-eafd-4a00-ac58-dacc9b705f42.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195312274-63ad3dbe-0447-4310-84a9-8ad644db820d.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195312312-3e145c49-22b1-4704-87f0-c25544e80558.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195312341-a6ab2415-c928-4582-bcb6-2000cd5cf40c.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195312382-54897e80-e8c7-48c9-97fe-4e2ae5e1d0ad.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195312417-eee07814-9512-4952-b156-4c864ee9cff0.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195317727-d13434c7-a0cb-4d8c-b938-938cd6b10459.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195317761-7c2386d9-5ff6-4067-b64b-9a824b4fb989.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195317788-0a0e1478-660e-4df7-87e7-0e9f650e5ac1.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195317818-82e045c9-2490-49e4-932a-64770a35d80d.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195317851-c71f9666-b64c-4551-bc90-53655472e3ef.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195317880-661793d6-67fd-4c25-aa5c-a2990c8b384d.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195318341-ba9e7913-d84f-4f0b-b5c5-8bcc294acca1.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195318385-d506493e-d2f3-4485-bd48-7cc2c33c52f3.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195318409-6344c874-7413-4420-92e8-bc1b585eac92.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195318470-606eb236-9aa7-4f29-9a5a-afd50f5ed0f7.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195318495-844d18c3-2cd5-4eda-b5a0-85dddac66467.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195318545-b87966b6-bab8-4bf0-a73d-ac7d85342225.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195318706-4f5fab19-f784-4b57-b0bb-47a04f8da58c.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195318739-70d82d36-ebea-4628-8127-d8e0d45804a6.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195318761-8fcb9dc0-2309-4b07-bf58-e36471a4334a.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/195318793-9ef8e7a4-fddd-4c64-9d72-bf56bf70a252.png)

**Comment by physics08** [comment]
[data.csv](https://github.com/physics08/physics08/files/9808786/data.csv)

**Comment by physics08** [comment]
[fruits.csv](https://github.com/physics08/physics08/files/9809024/fruits.csv)

**Comment by physics08** [comment]
[fruits2.csv](https://github.com/physics08/physics08/files/9809139/fruits2.csv)

**Comment by physics08** [comment]
[fruits3.csv](https://github.com/physics08/physics08/files/9809179/fruits3.csv)

**Comment by physics08** [comment]
[fruits3.csv](https://github.com/physics08/physics08/files/9809236/fruits3.csv)

**Comment by physics08** [comment]
[fruits3.csv](https://github.com/physics08/physics08/files/9809251/fruits3.csv)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/207566296-e245fe01-40e1-47c5-a24d-16718cfa73b1.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/207566470-544e506e-9d87-4341-90a3-00e1df1e004d.png)
![image](https://user-images.githubusercontent.com/105033906/207566502-f0d7dbb7-5a8a-4a15-acb4-d82431ab10f8.png)
![image](https://user-images.githubusercontent.com/105033906/207566534-940cf6ad-7242-4d53-909b-917b389fedf0.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/207566642-162e848f-bde4-441e-8b76-59b0fe1af9bc.png)


![image](https://user-images.githubusercontent.com/105033906/207566763-a76c69f4-fabf-4d22-8476-0837d8b4a44a.png)


![image](https://user-images.githubusercontent.com/105033906/207566783-3219598a-b9e0-4477-a50d-69693a65dcb3.png)

![image](https://user-images.githubusercontent.com/105033906/207566838-5543a35c-6f3b-450a-b64f-e5bea9921fff.png)

**Comment by physics08** [comment]
![VAE8](https://user-images.githubusercontent.com/105033906/208328205-1b8b0593-484a-439a-97bd-e278bef77594.jpg)
![vae6](https://user-images.githubusercontent.com/105033906/208328239-1e66934d-d5df-4fb0-a9f2-1be96ebd3ccb.png)
![vae1](https://user-images.githubusercontent.com/105033906/208328269-044aad02-086d-4bb2-af7d-3389dc41a5d2.png)
![VAE5](https://user-images.githubu
![vae7](https://user-images.githubusercontent.com/105033906/208328310-cbc2ffd0-fbee-4ca4-91e5-b57775c97158.png)
sercontent.com/105033906/208328287-eaf9880c-299c-4aea-9455-adb7526a8760.png)

![vae9](https://user-images.githubusercontent.com/105033906/208328345-fdbc141a-8508-4bd7-9f5f-ae1f85539d3e.jpg)

![vae10](https://user-images.githubusercontent.com/105033906/208328377-86242cd8-420a-4756-aa35-114842853602.png)
![vae11](https://user-images.githubusercontent.com/105033906/208328407-ea691be9-ce87-406c-b2e0-14fd5a37c7b0.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/208669093-c6d0c61e-3997-4865-ab20-e8251a630c7e.png)


![image](https://user-images.githubusercontent.com/105033906/208669106-c36c9139-873d-4696-9d96-c36658c6cd0a.png)



![image](https://user-images.githubusercontent.com/105033906/208669123-9bcf8354-e26c-44c5-9b9a-56f58adfa238.png)

![image](https://user-images.githubusercontent.com/105033906/208669143-869a3554-9e81-4350-8942-33da0ec7515e.png)


![image](https://user-images.githubusercontent.com/105033906/208669162-90ee5df0-cdd1-457a-a6e8-57e489c6250b.png)


![image](https://user-images.githubusercontent.com/105033906/208669177-bcfa4256-60f8-4195-856b-8bd9baddd83b.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/208686251-dd535a2c-b375-4923-aafd-dfc792415440.png)
![image](https://user-images.githubusercontent.com/105033906/208686258-2d6c4eae-f3ce-4a7c-a258-0ae93e626b76.png)

**Comment by physics08** [comment]
![image](https://user-images.githubusercontent.com/105033906/208836257-4ed02e02-bcba-4593-be15-8444e4f085d2.png)
